From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <herman@mailbox.org>
Date: Tue, 25 Aug 2026 08:00:00 +0200
Subject: [PATCH] lib/build_OID_registry: don't embed the build path in
 output

Kbuild invokes this script as "perl $(srctree)/lib/build_OID_registry",
so $0 is the absolute path to the script whenever the kernel is built
out-of-tree (O=), which is what OE/Yocto always does. That absolute
path, which contains ${WORKDIR}/${TMPDIR}, ends up in the "Automatically
generated by" comment of lib/oid_registry_data.c and trips the
"buildpaths" QA check on the -src package.

Strip the $srctree prefix (exported by the top-level Makefile) from $0
before printing it, same as upstream commit 0f586f4ee8ad ("lib/build_OID_registry:
Don't mention the full path of the script in output") does for
abs_srctree, which this kernel version does not export yet.

Upstream-Status: Backport [https://git.yoctoproject.org/linux-yocto/commit/?h=v5.15/standard/base&id=0f586f4ee8adacac79b64d1f3d47799a5eb7fbea]
Signed-off-by: Herman van Hazendonk <herman@mailbox.org>
---
 lib/build_OID_registry | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/build_OID_registry b/lib/build_OID_registry
index d7fc32ea8ac2..324ee6520ee2 100755
--- a/lib/build_OID_registry
+++ b/lib/build_OID_registry
@@ -39,7 +39,7 @@ close IN_FILE || die;
 #
 open C_FILE, ">$ARGV[1]" or die;
 print C_FILE "/*\n";
-print C_FILE " * Automatically generated by ", $0, ".  Do not edit\n";
+print C_FILE " * Automatically generated by ", $0 =~ s#^\Q$ENV{'srctree'}/\E##r, ".  Do not edit\n";
 print C_FILE " */\n";
 
 #
-- 
2.43.0
